velox/type/DecimalUtil.h

```diff
@@ -301,6 +302,79 @@ class DecimalUtil {
     return remainder * resultSign;
   }
 
+  /// Returns the max required size to convert the decimal of this precision and
+  /// scale to varchar. A varchar's size is estimated with unscaled value
+  /// digits, dot, leading zero, and possible minus sign.
+  static int32_t maxStringViewSize(int precision, int scale) {
+    int32_t rowSize = precision + 1; // Number and symbol.
+    if (scale > 0) {
+      ++rowSize; // A dot.
+    }
+    if (precision == scale) {
+      ++rowSize; // Leading zero.
+    }
+    return rowSize;
+  }
+
+  /// @brief Convert the unscaled value of a decimal to varchar and write to raw
+  /// string buffer from start position.
+  /// @tparam T The type of input value.
+  /// @param unscaledValue The input unscaled value.
+  /// @param scale The scale of decimal.
+  /// @param maxVarcharSize The estimated max size of a varchar.
+  /// @param startPosition The start position to write from.
+  /// @return A string view.
+  template <typename T>
+  static StringView convertToStringView(
+      T unscaledValue,
+      int32_t scale,
+      int32_t maxVarcharSize,
```

Comment:
Let's avoid mixing StringView and Varchar terms. Perhaps, rename to maxSize or consider replacing maxVarcharSize and startPosition parameters with a single std::string_view parameter.